sandbox/extract-imagery-metadata_mg.R

```diff
@@ -0,0 +1,181 @@
+
+# Between the BEGIN and END comments below, write code to extract the metadata attribute you're
+# working on. When you're done, you can wrap it in a function definition, taking only one parameter,
+# 'exif'. Here is an example of developing code to extract the number of images in an imagery
+# dataset.
+
+# BEGIN FUNCTION CODE
+
+# Get the number of images in the dataset by counting the rows of the EXIF data frame
+image_count = nrow(exif)
+
+# END FUNCTION CODE
+
+
+# Now here is an example of turning that code into a function
+
+extract_image_count = function(exif) {
+
+  # Get the number of images in the dataset by counting the rows of the EXIF data frame
+  image_count = nrow(exif)
+
+  return(image_count)
+
+}
+
+
+# Now you can test the function on the EXIF data
+
+image_count = extract_image_count(exif)
+image_count
+
+# Once it is working right, you can move this function to your
+# 'R/imagery-metadata-extraction_<initials>.R' file and then add a call to this function from within
+# your 'extract_metadata_<initials>' function. Once it is in there, then you can run the top part of
+# this script again, and when it extracts the metadata for each EXIF dataset, your additional
+# metadata should be included.
+
+
+#START CODE FOR CENTROID LAT/LONG
+
+GPSLocation = exif$GPSPosition
+
+#splitting the list (was this step really necessary for my dataset? I'm not really sure this actually helped?)
+coordinates <- strsplit(GPSLocation, split = ";")
+
+# Separating the GPS from the whole exif file
+almostdone <- strsplit(unlist(coordinates), split = ",")
+
+#Creating a seperate dataframe with just the coordinates to call upon later
+isolatedcoordinates <- as.data.frame(do.call(rbind, almostdone))
+
+library(stringr)
+#Need to download this package as it will help with the following functions
+separator(df)
+
+install.packages("tidyverse")
+library(tidyverse)
+
+df2 = separate(exif,
+        col = GPSPosition,
+        into = c("lat", "long"),
+        sep = " " )
+#Here this worked, instead of doing by number of characters, did it by seperating at
+#the space in between the two (since the number of characters differed from row to row).
+
+print(df2)
+#printed out the new df that includes the "lat" and "long"
+
+#when looking online, it says that the way to find the centroid lat/long is to
+#just use the average of the lat and the average of the long
+
+str(df2$lat)
+str(df2$long)
+
+#here checking what type of character the lat and long are within the
+#dataset, if it is not numeric, will need to change it.
+
+df2$lat = as.numeric(df2$lat)
+df2$long = as.numeric(df2$long)
+
+#had to convert to numeric numbers since it was thinking that the column was a
+#character.
+
+xcoordinate = mean(df2$lat, na.rm=TRUE)
+ycoordinate = mean(df2$long, na.rm= TRUE)
+
+#finding the mean of the lat and long, since this is all one mission,
+#it should give us the centroid lat/long, at least from what I researched.
+
+print(ycoordinate)
+print(xcoordinate)
```

Comment:
Once you have computed the centroid as a `sf` object, you an simply return that object, rather than two objects containing the text of the coordinates.